dwave/optimization/mathematical.py

```diff
@@ -428,6 +433,131 @@ def hstack(arrays: collections.abc.Sequence[ArraySymbol]) -> ArraySymbol:
         return concatenate(arrays, 1)
 
 
+class LPResult:
+    """The outputs of a linear program run."""
+
+    lp: LP
+    """The LP symbol."""
+
+    def __init__(self, lp: LP):
+        self.lp = lp
+
+    @functools.cached_property
+    def fun(self) -> LPObjectiveValue:
+        """The value of the objective as an array symbol."""
+        return LPObjectiveValue(self.lp)
+
+    @functools.cached_property
+    def success(self) -> LPFeasible:
+        """``True`` is the linear program found the optimal value as an array symbol."""
+        return LPFeasible(self.lp)
+
+    @functools.cached_property
+    def x(self) -> LPSolution:
+        """The assignments to the decision variables as an array symbol."""
+        return LPSolution(self.lp)
+
+
+def linprog(
+        c: ArraySymbol,
+        A_ub: typing.Optional[ArraySymbol] = None,  # alias for A
+        b_ub: typing.Optional[ArraySymbol] = None,
+        A_eq: typing.Optional[ArraySymbol] = None,
+        b_eq: typing.Optional[ArraySymbol] = None,
+        *,  # the args up until here match SciPy's linprog() which accepts them positionally
+        b_lb: typing.Optional[ArraySymbol] = None,
+        A: typing.Optional[ArraySymbol] = None,
+        lb: typing.Optional[ArraySymbol] = None,
+        ub: typing.Optional[ArraySymbol] = None,
+        ) -> LPResult:
+    r"""Solve a :term:`linear program` defined by the input array symbol(s).
+
+    Linear programs solve problems of the form:
+
+    .. math::
+        \text{minimize:} \\
+        & c^T x \\
+        \text{subject to:} \\
+        b_{lb} &\leq A x &\leq b_{ub} \\
+        b_{eq} &= A_{eq} x \\
+        l &\leq x &\leq u
```

Comment:
Perhaps we should use `lb` and `ub` here instead of `l` and `u`?

Reply:
And in the example below

Reply:
It messed up the sphinx math formatting unfortunately. So this is a compromise